### PR TITLE
Remove exception message from AnyException

### PR DESCRIPTION
**Pull request by cocreature**
This doesn’t really make that much sense in there and it makes some
things like Daml Script harder because I need to duplicate the logic
to get the exception message.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala [hunk 1]
```diff
@@ -1562,14 +1562,23 @@ private[lf] object SBuiltin {
         machine: Machine,
     ): Unit = {
       args.get(0) match {
-        case SAnyException(_, messageFunction, innerValue) =>
-          machine.ctrl = SEApp(messageFunction, Array(SEValue(innerValue)))
+        case SAnyException(ty, innerValue) =>
+          machine.ctrl = SEApp(exceptionMessage(ty), Array(SEValue(innerValue)))
```
Comment (cocreature) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala:1566]: I don’t like computing this at runtime all that much. I’m not really worried about performance and it seems nicer than what we had before but if someone has a better proposal I’m all ears.

Comment (sofiafaro-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala:1566]: This function should be called really infrequently, so I think it should be fine.

**Review by sofiafaro-da — APPROVED** [review]
This change makes sense. Thank you!

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala (hunk 1 above)
Comment (nickchapman-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala:1566]: I agree. Can't imagine there is any perf issue here.

**Review by nickchapman-da — APPROVED** [review]
LGTM

**Pull request merged**